File: Java Klassen/src/model/plan/BuildRoad.java
package model.plan;

import model.RoadBuildPlace;

/**
 * Klasse, die die Aktion das Bauen von Straﬂen darstellt
 * 
 * @author Tjark Harjes
 */
public class BuildRoad extends Action {

	private int row, column;
	
	public BuildRoad() {
	}

	public BuildRoad(RoadBuildPlace randomRoadPlace) {
		this.row = randomRoadPlace.row;
		this.column = randomRoadPlace.column;
	}
	
	public int getRow() {
		return row;
	}


	public void setRow(int row) {
		this.row = row;
	}


	public int getColumn() {
		return column;
	}


	public void setColumn(int column) {
		this.column = column;
	}
}

File: Unity Projekt/Assets/Scripts/CBR.Plan/BuildRoad.cs
using Assets.Scripts.Model;
using System.Runtime.Serialization;
namespace Assets.Scripts.CBR.Plan
{
    /**
     * Klasse, die die Aktion das Bauen von Straßen darstellt
     */
    [DataContract]
    public class BuildRoad : Action
    {

        /**
        * Attribut zur Speicherung, in welcher Zeile gebaut werden soll 
        */
        [DataMember]
        public int row;

        /**
         * Attribut zur Speicherung, in welcher Spalte gebaut werden soll 
         */
        [DataMember]
        public int column;


        public BuildRoad(int row, int column) : base("BuildRoad")
        {
            this.row = row;
            this.column = column;
        }


        public BuildRoad() : base("BuildRoad")
        {
        }
    }
}
